server/src/main/java/org/elasticsearch/common/util/StringLiteralDeduplicator.java

```diff
@@ -20,8 +18,6 @@
  */
 public final class StringLiteralDeduplicator {
 
-    private static final Logger logger = LogManager.getLogger(StringLiteralDeduplicator.class);
```

Comment:
We use this thing before the logging is fully initialized now in some cases leading to the same issue we resolve by indirection in `org.elasticsearch.common.settings.Settings.DeprecationLoggerHolder`. I don't think the logging here is valuable enough to go for any kind of sophisticated solution to keep it around so I just dropped it.